solidity/dashboard/src/components/Timer.jsx

```diff
@@ -1,8 +1,82 @@
-import React from "react"
+import React, { useEffect, useState } from "react"
+import moment from "moment";
 
-const Timer = ({ target }) => {
-  // TODO implement timer
-  return <>{target}</>
+const Timer = ({ targetInUnix }) => {
+  const [remainingDays, setRemainingDays] = useState(0)
+  const [remainingHours, setRemainingHours] = useState(0)
+  const [remainingMinutes, setRemainingMinutes] = useState(0)
+  const [remainingSeconds, setRemainingSeconds] = useState(0)
+
+  useEffect(() => {
+    const myInterval = setInterval(() => {
+      if (remainingSeconds > 0) {
+        setRemainingSeconds(remainingSeconds - 1)
+      }
+      if (remainingSeconds === 0) {
+        if (
+          remainingDays === 0 &&
+          remainingHours === 0 &&
+          remainingMinutes === 0
+        ) {
+          clearInterval(myInterval)
+        } else {
+          setRemainingSeconds(59)
+          if (remainingMinutes > 0) {
+            setRemainingMinutes(remainingMinutes - 1)
+          } else if (remainingMinutes === 0) {
+            setRemainingMinutes(59)
+            if (remainingHours > 0) {
+              setRemainingHours(remainingHours - 1)
+            } else if (remainingHours === 0) {
+              setRemainingHours(23)
+              if (remainingDays > 0) {
+                setRemainingDays(remainingDays - 1)
+              }
+            }
+          }
+        }
+      }
+    }, 1000)
+    return () => {
+      clearInterval(myInterval)
+    }
+  })
+
+  useEffect(() => {
```

Comment:
purely cosmetic thing. Could we set this hook as the first? It'll help to understand why we use `0` in `useState` hook.